ford/sourceform.py

```diff
@@ -931,22 +965,28 @@ def __init__(
                         # expression doesn't catch that so we first rule such a
                         # GOTO out.
                         if not self.ARITH_GOTO_RE.search(line):
-                            callvals = self.CALL_RE.findall(line)
+                            callvals = []
+                            matches = self.CALL_RE.finditer(line)
+                            for match in matches:
+                                callvals.append(get_call_chain(match))
                             for val in callvals:
                                 if (
-                                    val.lower() not in self.calls
-                                    and val.lower() not in INTRINSICS
+                                    not any(d["name"] == val["name"] for d in self.calls)
```

Comment:
Please could add a comment here? This logic is now getting a bit complex!

In fact, it looks like this logic is duplicated below -- probably worth pulling out into a named helper function. I find naming things like this helps a lot with understanding the code